AIX test failure on zero disk capacity

```
Failures (1):
  JUnit Jupiter:PerfstatTest:testQueryDiskStats()
    MethodSource [className = 'oshi.driver.unix.aix.perfstat.PerfstatTest', methodName = 'testQueryDiskStats', methodParameterTypes = '']
    => java.lang.AssertionError: Should have a nonzero disk capacity
Expected: a value greater than <0L>
     but: <0L> was equal to <0L>
       org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
       oshi.driver.unix.aix.perfstat.PerfstatTest.testQueryDiskStats(PerfstatTest.java:58)
```
This is probably the "Virtual SCSI Optical" drives `cd0` and `cd1`:
```
Disks:
 hdisk0: (model: Virtual SCSI Disk Drive - S/N: unknown) size: 66.5 GB, reads: 521593 (4.6 GiB), writes: 5739750 (51.0 GiB), xfer: 4927635
 |-- hd5: hd5 (boot) Maj:Min=10:1, size: 67.1 MB
 |-- hd6: hd6 (paging) Maj:Min=10:2, size: 2.1 GB
 |-- hd8: hd8 (jfs2log) Maj:Min=10:3, size: 67.1 MB
 |-- hd4: hd4 (jfs2) Maj:Min=10:4, size: 2.1 GB @ /
 |-- hd2: hd2 (jfs2) Maj:Min=10:5, size: 3.2 GB @ /usr
 |-- hd9var: hd9var (jfs2) Maj:Min=10:6, size: 3.0 GB @ /var
 |-- hd3: hd3 (jfs2) Maj:Min=10:7, size: 3.0 GB @ /tmp
 |-- hd1: hd1 (jfs2) Maj:Min=10:8, size: 67.1 MB @ /homehd1
 |-- hd10opt: hd10opt (jfs2) Maj:Min=10:9, size: 4.3 GB @ /opt
 |-- hd11admin: hd11admin (jfs2) Maj:Min=10:10, size: 134.2 MB @ /admin
 |-- lg_dumplv: lg_dumplv (sysdump) Maj:Min=10:11, size: 2.1 GB
 |-- livedump: livedump (jfs2) Maj:Min=10:12, size: 268.4 MB @ /var/adm/ras/livedump
 hdisk1: (model: Virtual SCSI Disk Drive - S/N: unknown) size: 68.6 GB, reads: 22698 (274.3 MiB), writes: 10678210 (126.0 GiB), xfer: 8728493
 |-- lgscratch: lgscratch (jfs2log) Maj:Min=37:1, size: 67.1 MB
 |-- lvscratch: lvscratch (jfs2) Maj:Min=37:2, size: 68.5 GB @ /scratch
 hdisk4: (model: Virtual SCSI Disk Drive - S/N: unknown) size: 66.5 GB, reads: 149331 (3.3 GiB), writes: 525376 (11.5 GiB), xfer: 780075
 |-- hd01: hd01 (jfs2) Maj:Min=38:1, size: 56.4 GB @ /home
 hdisk9: (model: Virtual SCSI Disk Drive - S/N: unknown) size: 66.5 GB, reads: 172418 (4.2 GiB), writes: 558540 (13.7 GiB), xfer: 983601
 |-- hd01: hd01 (jfs2) Maj:Min=38:1, size: 56.4 GB @ /home
 hdisk8: (model: Virtual SCSI Disk Drive - S/N: unknown) size: 66.5 GB, reads: 214824 (4.5 GiB), writes: 561992 (11.9 GiB), xfer: 942904
 |-- hd01: hd01 (jfs2) Maj:Min=38:1, size: 56.4 GB @ /home
 hdisk7: (model: Virtual SCSI Disk Drive - S/N: unknown) size: 66.5 GB, reads: 257362 (6.5 GiB), writes: 596715 (15.0 GiB), xfer: 936016
 |-- hd01: hd01 (jfs2) Maj:Min=38:1, size: 56.4 GB @ /home
 hdisk6: (model: Virtual SCSI Disk Drive - S/N: unknown) size: 66.5 GB, reads: 209945 (4.3 GiB), writes: 551727 (11.3 GiB), xfer: 875359
 |-- hd01: hd01 (jfs2) Maj:Min=38:1, size: 56.4 GB @ /home
 hdisk3: (model: Virtual SCSI Disk Drive - S/N: unknown) size: 66.5 GB, reads: 195302 (4.6 GiB), writes: 578535 (13.7 GiB), xfer: 893181
 |-- hd01: hd01 (jfs2) Maj:Min=38:1, size: 56.4 GB @ /home
 hdisk2: (model: Virtual SCSI Disk Drive - S/N: unknown) size: 66.5 GB, reads: 742661 (4.5 GiB), writes: 5530228 (33.8 GiB), xfer: 6428364
 |-- hd01: hd01 (jfs2) Maj:Min=38:1, size: 56.4 GB @ /home
 |-- log08: log08 (jfs2log) Maj:Min=38:2, size: 33.6 MB
 hdisk5: (model: Virtual SCSI Disk Drive - S/N: unknown) size: 66.5 GB, reads: 180974 (4.3 GiB), writes: 505037 (11.9 GiB), xfer: 720967
 |-- hd01: hd01 (jfs2) Maj:Min=38:1, size: 56.4 GB @ /home
 cd1: (model: Virtual SCSI Optical Served by VIO Server - S/N: unknown) size: ?, reads: ? (?), writes: ? (?), xfer: ?
 cd0: (model: Virtual SCSI Optical Served by VIO Server - S/N: unknown) size: ?, reads: ? (?), writes: ? (?), xfer: ?
```
I may relax the test to allow 0 in the case there are no partitions.

--- a/.github/workflows/aix.yaml
+++ b/.github/workflows/aix.yaml
@@ -37,7 +37,7 @@ jobs:
           rm -fR ~/.cache/JNA
           rm -fR ~/javashared*
           cd ~/git/oshi
-          export JAVA_HOME=/opt/ibm-semeru-open-17-jdk
+          export JAVA_HOME=~/java/jdk-17.0.8.1+1
           export PATH=$PATH:/opt/freeware/bin
           export MAVEN_OPTS="-Xshareclasses:none"
           git checkout master

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,7 @@
 # 6.4.7 (in progress)
 
-* Your contribution here!
+##### Bug fixes / Improvements
+* [#2460](https://github.com/oshi/oshi/pull/2460): Fix AIX tests for virtual/unused drives - [@dbwiddis](https://github.com/dbwiddis).
 
 # 6.4.0 (2022-12-02), 6.4.1 (2023-03-18), 6.4.2 (2023-05-02), 6.4.3 (2023-06-06), 6.4.4 (2023-07-01), 6.4.5 (2023-08-20), 6.4.6 (2023-09-24)
 

--- a/oshi-core/src/test/java/oshi/driver/unix/aix/perfstat/PerfstatTest.java
+++ b/oshi-core/src/test/java/oshi/driver/unix/aix/perfstat/PerfstatTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2022 The OSHI Project Contributors
+ * Copyright 2022-2023 The OSHI Project Contributors
  * SPDX-License-Identifier: MIT
  */
 package oshi.driver.unix.aix.perfstat;
@@ -7,6 +7,7 @@ package oshi.driver.unix.aix.perfstat;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.emptyString;
 import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 import static org.hamcrest.Matchers.not;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
@@ -55,7 +56,12 @@ class PerfstatTest {
         perfstat_disk_t[] disks = PerfstatDisk.queryDiskStats();
         assertThat("Should have at least one disk", disks.length, greaterThan(0));
         for (perfstat_disk_t disk : disks) {
-            assertThat("Should have a nonzero disk capacity", disk.size, greaterThan(0L));
+            // Virtual disks may give 0 capacity but also have 0 time
+            if (disk.time == 0) {
+                assertThat("Should have a nonnegative disk capacity", disk.size, greaterThanOrEqualTo(0L));
+            } else {
+                assertThat("Should have a nonzero disk capacity", disk.size, greaterThan(0L));
+            }
         }
     }
 

